WalletWasabi/Wallets/WalletFilterProcessor.cs

```diff
@@ -265,14 +268,12 @@ private async void ReorgedAsync(object? sender, FilterModel invalidFilter)
 
 			using (await ReorgLock.LockAsync(CancellationToken.None).ConfigureAwait(false))
 			{
-				KeyManager.SetMaxBestHeight(new Height(invalidFilter.Header.Height - 1));
+				Height newBestHeight = new(invalidFilter.Header.Height - 1);
+				KeyManager.SetMaxBestHeight(newBestHeight);
 				TransactionProcessor.UndoBlock((int)invalidFilter.Header.Height);
 				BitcoinStore.TransactionStore.ReleaseToMempoolFromBlock(invalidBlockHash);
 
-				if (BlockProvider is SmartBlockProvider smartBlockProvider)
-				{
-					await smartBlockProvider.RemoveAsync(invalidBlockHash, CancellationToken.None).ConfigureAwait(false);
-				}
+				await BlockDownloadService.RemoveBlocksAsync((uint)newBestHeight.Value).ConfigureAwait(false);
```

Comment:
Are we sure we want to remove `block number = invalidFilter.Header.Height - 1` ?

I thought we wanted to remove `block number =  invalidFilter.Header.Height`, or do I understand Reorg incorrectly?